### PR TITLE
`Lectures`: Add lecture unit attachment file size pre-check

### PR DESCRIPTION
**Pull request by pal03377**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

There was some confusion about uploading lecture unit attachments when the file size was too big.
Currently, you have to wait for the upload to then receive a generic HTTP error 400. This is slightly annoying and overall no good user experience.

### Description

Before you try to upload a lecture unit attachment with a size larger than the maximum (20 MB), you get an error message and cannot create the lecture unit.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course
- A file larger than 20 MB, e.g. [21M.zip](https://github.com/ls1intum/Artemis/files/11290436/21M.zip)
- A file smaller than 20 MB, e.g. [19M.zip](https://github.com/ls1intum/Artemis/files/11290441/19M.zip)

1. Log in to Artemis
2. Navigate to Course Administration > Your Course > Lectures
3. Create a lecture on the top right
4. Go to the `Lecture Units`
5. Choose "File", then enter a name
6. Try to upload a file larger than 20 MB. An error message should appear and the submit button should be disabled.
7. Verify that uploading a file smaller than 20 MB works.

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| attachment-unit-form.component.ts | 97.77% | ✅              |

### Screenshots

screenshot-2023-04-20_001252

screenshot-2023-04-20_001253

### CODE REVIEW

File: src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-unit-form/attachment-unit-form.component.ts [hunk 1]
```diff
@@ -61,6 +62,7 @@ export class AttachmentUnitFormComponent implements OnInit, OnChanges {
     file: File;
     fileName?: string;
     fileInputTouched = false;
+    fileTooBig: boolean;
```
Comment (laadvo) [inline src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-unit-form/attachment-unit-form.component.ts:65]:
```suggestion
    isFileTooBig: boolean;
```

**Review by laadvo — COMMENTED** [review]
The code looks good, I just have a small suggestion regarding the naming

**Review by JohannesStoehr — APPROVED** [review]
Code changes and test look god to me

**Review by DominikRemo — APPROVED** [review]
tested on ts1. works as expected

**Review by bassner — APPROVED** [review]
LGTM

**Pull request merged**